api/python/t4/data_transfer.py

```diff
@@ -531,3 +531,27 @@ def get_bytes(src):
     else:
         raise NotImplementedError
     return data, meta
+
+def get_meta(src):
+    """
+    Gets S3 metadata for the object at a given S3 URI.
+    """
+    bucket, key, version_id = parse_s3_url(urlparse(src))
+    params = {
+        'Bucket': bucket,
+        'Key': key
+    }
+    if version_id:
+        params['VersionId'] = version_id
+    resp = s3_client.head_object(**params)
+    meta = _parse_metadata(resp)
+    return meta
+
+def set_meta(dest, meta):
```

Comment:
This one doesn't touch low-level S3 stuff, and just calls other helpers - so I'd rather delete it and move the code into `Bucket.set_meta`.

(Also, nothing else in `data_transfer.py` even knows about "user_meta", so doesn't seem like the right level of abstraction here.)